satellite/metainfo/metainfo.go

```diff
@@ -702,19 +707,55 @@ func (endpoint *Endpoint) CreateBucket(ctx context.Context, req *pb.BucketCreate
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	bucket, err := convertProtoToBucket(req, keyInfo.ProjectID)
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, err.Error())
-	}
+	bucket, err := endpoint.metainfo.GetBucket(ctx, req.GetName(), keyInfo.ProjectID)
+	if err == nil {
+		var partnerID uuid.UUID
+		err = partnerID.UnmarshalJSON(req.GetPartnerId())
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, err.Error())
+		}
 
-	bucket, err = endpoint.metainfo.CreateBucket(ctx, bucket)
-	if err != nil {
-		return nil, Error.Wrap(err)
+		// partnerID not set
+		if partnerID.IsZero() {
+			return resp, status.Errorf(codes.AlreadyExists, "Bucket already exists")
+		}
+
+		//update the bucket
+		bucket.PartnerID = partnerID
```

Comment:
should we check if the bucket already has this partnerID set? for example, if bucket.Partner == partnerID, then return

Reply:
There is checkBucketAttribution() before updateBucket checks if that bucket has already an attribution done. If it done, it returns error even before going to next line too updateBucket. an attribution has to exist even before the bucket is created. So it is handled.